flink-tests/src/test/java/org/apache/flink/runtime/metrics/JobManagerMetricsITCase.java

```diff
@@ -0,0 +1,221 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.metrics;
+
+import org.apache.flink.client.program.MiniClusterClient;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.metrics.Gauge;
+import org.apache.flink.metrics.MetricConfig;
+import org.apache.flink.metrics.reporter.AbstractReporter;
+import org.apache.flink.runtime.execution.Environment;
+import org.apache.flink.runtime.jobgraph.JobGraph;
+import org.apache.flink.runtime.jobgraph.JobStatus;
+import org.apache.flink.runtime.jobgraph.JobVertex;
+import org.apache.flink.runtime.jobgraph.tasks.AbstractInvokable;
+import org.apache.flink.test.util.MiniClusterResource;
+import org.apache.flink.test.util.MiniClusterResourceConfiguration;
+
+import org.junit.Assume;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import static org.apache.flink.configuration.MetricOptions.REPORTERS_LIST;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * Integration tests for proper initialization of the job manager metrics.
+ */
+public class JobManagerMetricsITCase {
+
+	private static final String JOB_MANAGER_METRICS_PREFIX = "localhost.jobmanager.";
+	private static final String TASK_SLOTS_AVAILABLE = "taskSlotsAvailable";
+	private static final String TASK_SLOTS_TOTAL = "taskSlotsTotal";
+	private static final String NUM_REGISTERED_TASK_MANAGERS = "numRegisteredTaskManagers";
+	private static final String NUM_RUNNING_JOBS = "numRunningJobs";
+
+	private static CountDownLatch invokeLatch;
+
+	private MiniClusterClient clusterClient;
+	private JobGraph jobGraph;
+
+	@ClassRule
+	public static final MiniClusterResource MINI_CLUSTER_RESOURCE = new MiniClusterResource(
+		new MiniClusterResourceConfiguration.Builder()
+			.setConfiguration(getConfiguration())
+			.setNumberTaskManagers(1)
+			.setNumberSlotsPerTaskManager(1)
+			.build());
+
+	@Before
+	public void setUp() throws Exception {
+		invokeLatch = new CountDownLatch(1);
+
+		Assume.assumeTrue(
+			"ClusterClient is not an instance of MiniClusterClient",
+			MINI_CLUSTER_RESOURCE.getClusterClient() instanceof MiniClusterClient);
+
+		clusterClient = (MiniClusterClient) MINI_CLUSTER_RESOURCE.getClusterClient();
+		clusterClient.setDetached(true);
```

Comment:
You could simplify this section by simple defining a job that has a blocking UDF and executing it in a separate thread.